go/chat/push.go

```diff
@@ -1010,6 +1010,50 @@ func (g *PushHandler) SetConvSettings(ctx context.Context, m gregor.OutOfBandMes
 	return nil
 }
 
+func (g *PushHandler) SubteamRename(ctx context.Context, m gregor.OutOfBandMessage) (err error) {
+	defer g.Trace(ctx, func() error { return err }, "SubteamRename")()
+	if m.Body() == nil {
+		return errors.New("gregor handler for chat.subteamRename: nil message body")
+	}
+
+	var update chat1.SubteamRenameUpdate
+	reader := bytes.NewReader(m.Body().Bytes())
+	dec := codec.NewDecoder(reader, &codec.MsgpackHandle{WriteExt: true})
+	if err = dec.Decode(&update); err != nil {
+		return err
+	}
+	uid := gregor1.UID(m.UID().Bytes())
+
+	// Order updates based on inbox version of the update from the server
+	cb := g.orderer.WaitForTurn(ctx, uid, update.InboxVers)
+	bctx := BackgroundContext(ctx, g.G())
+	go func(ctx context.Context) {
+		defer g.Trace(ctx, func() error { return nil }, "SubteamRename(goroutine)")()
+		<-cb
+		g.Lock()
+		defer g.Unlock()
+		defer g.orderer.CompleteTurn(ctx, uid, update.InboxVers)
+		// Get and localize the conversation to get the new tlfname.
+		inbox, err := g.G().InboxSource.Read(ctx, uid, nil, true, &chat1.GetInboxLocalQuery{
+			ConvIDs: update.ConvIDs,
+		}, nil)
+		if err != nil {
+			g.Debug(ctx, "resolve: unable to read conversation: %s", err.Error())
+			return
+		}
+		if len(inbox.Convs) != len(update.ConvIDs) {
+			g.Debug(ctx, "resolve: unable to find all conversations")
+		}
+
+		for _, conv := range inbox.Convs {
```

Comment:
Why not just send up a list?

Reply:
we could but we'd have to separate each batch to a single topic type so i thought this would be easier

Reply:
Think we should send up a list, isn't there some common code for topic type stuff? I just worry when the UI gets like a stream of little updates. It usually results in a bunch of little RPC calls into the service which can hurt perf.